buildSrc/src/main/java/org/elasticsearch/gradle/precommit/PrecommitTask.java

```diff
@@ -31,12 +33,16 @@
 
     @OutputFile
     public File getSuccessMarker() {
-        return new File(getProject().getBuildDir(), "markers/" + this.getName());
+        return new File(getProjectLayout().getBuildDirectory().getAsFile().get(), "markers/" + this.getName());
     }
 
     @TaskAction
     public void writeMarker() throws IOException {
         Files.write(getSuccessMarker().toPath(), new byte[] {}, StandardOpenOption.CREATE);
     }
 
+    @Inject
+    protected ProjectLayout getProjectLayout() {
```

Comment:
Where did we land on using method injection vs constructor injection? Should we standardize on one vs the other?

Reply:
I think by default we agreed on ctor injection. In this case I went with method injection due to keeping this local as this is an abstract task and all its descends would need to pass the injected service down here